sql/parser/eval.go

```diff
@@ -1161,7 +1165,11 @@ func (ctx EvalContext) evalCastExpr(expr *CastExpr) (Datum, error) {
 		case DString:
 			return ParseDate(d)
 		case DTimestamp:
-			return DDate{Time: d.Truncate(24 * time.Hour)}, nil
+			loc, err := ctx.GetLocation()
+			if err != nil {
+				return DNull, err
+			}
+			return MakeDDate(d.Time.In(loc)), nil
```

Comment:
yes, this is covered by driver_test; see the comment i left containing "verkakte"